app.py

```diff
@@ -21,41 +22,53 @@
     authority=app.config.get("AUTHORITY"),
     client_id=app.config["CLIENT_ID"],
     client_credential=app.config["CLIENT_SECRET"],
-    )
+)
+
 
 @app.route("/login")
 def login():
-    return render_template("login.html", version=identity.__version__, **auth.log_in(
-        scopes=app_config.SCOPE,  # Have user consent scopes during log-in
-        redirect_uri=url_for("auth_response", _external=True),  # Optional. If present, this absolute URL must match your app's redirect_uri registered in Azure Portal
-        ))
+    result = auth.log_in(
+        # Have user consent scopes during log-in
+        scopes=app_config.SCOPE,
+        # Optional. If present, this absolute URL must match your app's redirect_uri registered in Azure Portal
+        redirect_uri=url_for("auth_response", _external=True),
+    )
+    return render_template("login.html", version=identity.__version__, **result)
+
 
 @app.route(app_config.REDIRECT_PATH)
 def auth_response():
     result = auth.complete_log_in(request.args)
-    return render_template("auth_error.html", result=result) if "error" in result else redirect(url_for("index"))
+    if "error" not in result:
+        return redirect(url_for("index"))
+    return render_template("auth_error.html", result=result)
+
 
 @app.route("/logout")
 def logout():
     return redirect(auth.log_out(url_for("index", _external=True)))
 
+
 @app.route("/")
 def index():
     if not auth.get_user():
         return redirect(url_for("login"))
     return render_template('index.html', user=auth.get_user(), version=identity.__version__)
 
+
 @app.route("/call_downstream_api")
 def call_downstream_api():
     token = auth.get_token_for_user(app_config.SCOPE)
     if "error" in token:
         return redirect(url_for("login"))
-    api_result = requests.get(  # Use token to call downstream api
+    # Use access token to call downstream api
+    api_result = requests.get(
         app_config.ENDPOINT,
         headers={'Authorization': 'Bearer ' + token['access_token']},
-        ).json()
+        timeout=30
```

Comment:
Pylance recommends adding timeout to requests.get(), otherwise it's an infinite timeout that causes server to hang.

Reply:
Fair enough. I would suggest adding a trailing comma here, then.

```python
timeout=30,
```